### DISCUSSION TITLE
How to insert aside notation for a code generated table?

### DISCUSSION THREAD

**Discussion by aborruso** [quarto-dev/quarto-cli#2895] [Q&A] [answered]
Hi,
is there a way to insert an [aside notation](https://quarto.org/docs/authoring/article-layout.html#asides) for a code generated table?

The only way I have found is to use a margin content `{.column-margin}`.

Thank you

![image](https://user-images.githubusercontent.com/30607/196275997-3203c220-aa06-420d-9ea3-79f52d9bcb65.png)

**Comment by dragonstyle** [comment, marked as answer]
You can also use the class `aside` something like:

````markdown
---
title: Test Document
format: html
---


```{r}
#| warning: false
#| echo: false  
library(dplyr)
library(scales)
library(gt)

#Load data and arrange in descending order of emissions
#Generate a gt table from head of data
head(mtcars) %>% 
  gt()

```

::: {.aside}
This is an aside regarding the table.
:::

````

**Reply by aborruso** [reply]
Thank you, it works.

I wasn't able to get it to work, because I was reading [the guide](https://quarto.org/docs/authoring/article-layout.html#asides), I was using the sample code, and it didn't work.

Is it right the guide? Is right that code? If no, I can open a PR.

Thank you

```
[This is a span that has the class aside which places it in the margin without a footnote number.]{.aside}
```



![image](https://user-images.githubusercontent.com/30607/196505837-0b6d4bd5-bea5-43d9-bf79-257e080194d2.png)

**Reply by mcanouil** [reply]
"fenced div" (@dragonstyle version) is or should be similar to the span specification (see <https://quarto.org/docs/authoring/markdown-basics.html#divs-and-spans> for details).
If it's not working, it might be a bug.

**Reply by aborruso** [reply]
hi @mcanouil if I use this syntax, the aside text is not on the right of the text that precedes it, but at the bottom right of an empty line.

```
[This is a span that has the class aside which places it in the margin without a footnote number.]{.aside}
```

Instead If I use this, the aside text is on the right of the text that precedes it. It seems to me a bug.

Am I right?

```
::: {.aside}
This is an aside regarding the table.
:::
```

![image](https://user-images.githubusercontent.com/30607/196658644-cdd3bd54-5a30-4bd5-a4b8-7551b418a984.png)

**Reply by mcanouil** [reply]
spans are usually used for in-line styling while div are more like paragraph styling to simplify.
Here, it seems Quarto does not handle both types the same way.
I do not know if it's intended or not (thus a bug).

**Reply by dragonstyle** [reply]
It does seem surprising that we're treating the two so differently (and if anything you'd sort of expect the `div` to have block like behavior and the `span` more inline like behavior). I'll take a closer look!

**Reply by aborruso** [reply]
@dragonstyle thank you. 

For me the problem is that in the guide you have the aside put on the right, but if I use it I do no have it on the right.
Then if I'm not wrong there may be an inconsistency between documentation and results.

![image](https://user-images.githubusercontent.com/30607/196728615-47803be7-5dbd-493f-8abc-8a4f216f7b19.png)
